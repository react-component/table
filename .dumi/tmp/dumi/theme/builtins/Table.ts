// @ts-nocheck
// This file is generated by Umi automatically
// DO NOT CHANGE IT MANUALLY!
export { default } from '/Users/jilin/projects/antd/rc-table/node_modules/_dumi@2.0.17@dumi/theme-default/builtins/Table/index.js';
