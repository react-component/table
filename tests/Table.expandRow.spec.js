/* eslint-disable no-undef */
import React from 'react';
import { render, mount } from 'enzyme';
import { renderToJson, mountToJson } from 'enzyme-to-json';
import Table from '../src';

describe('Table.expand', () => {
  const columns = [
    { title: 'Name', dataIndex: 'name', key: 'name' },
    { title: 'Age', dataIndex: 'age', key: 'age' },
  ];
  const data = [
    { key: 0, name: 'Lucy', age: 27 },
    { key: 1, name: 'Jack', age: 28 },
  ];
  const expandedRowRender = () => <p>extra data</p>;
  const createTable = (props) => (
    <Table
      columns={columns}
      data={data}
      {...props}
    />
  );

  it('renders expand row correctly', () => {
    const wrapper = render(createTable({ expandedRowRender }));
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('renders tree row correctly', () => {
    const localData = [
      { key: 0, name: 'Lucy', age: 27, children: [
        { key: 2, name: 'Jim', age: 1 },
      ] },
      { key: 1, name: 'Jack', age: 28 },
    ];
    const wrapper = render(createTable({ data: localData }));
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('renders expand icon as cell', () => {
    const wrapper = render(createTable({
      expandedRowRender,
      expandIconAsCell: true,
    }));
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('renders expand icon to the specify column', () => {
    const wrapper = render(createTable({
      expandedRowRender,
      expandIconColumnIndex: 1,
    }));
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('renders nested data correctly', () => {
    const localData = [
      {
        key: '0',
        name: 'Lucy',
        age: 27,
        children: [
          { key: '0-1', name: 'Jim', age: '2' },
        ],
      },
      {
        key: 1,
        name: 'Jack',
        age: 28,
      },
    ];
    const wrapper = render(createTable({ data: localData }));
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('expand all rows by default', () => {
    const wrapper = render(createTable({
      expandedRowRender,
      defaultExpandAllRows: true,
    }));
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('expand rows by defaultExpandedRowKeys', () => {
    const wrapper = render(createTable({
      expandedRowRender,
      defaultExpandedRowKeys: [1],
    }));
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('controlled by expandedRowKeys', () => {
    const wrapper = mount(createTable({
      expandedRowRender,
      expandedRowKeys: [0],
    }));
    expect(mountToJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ expandedRowKeys: [1] });
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('renders expend row class correctly', () => {
    const expandedRowClassName = jest.fn().mockReturnValue('expand-row-test-class-name');
    const wrapper = render(createTable({
      expandedRowRender,
      expandedRowKeys: [0],
      expandedRowClassName,
    }));
    expect(renderToJson(wrapper)).toMatchSnapshot();
    expect(expandedRowClassName).toBeCalledWith(data[0], 0, 0);
  });

  it('fires expand change event', () => {
    const onExpand = jest.fn();
    const wrapper = mount(createTable({
      expandedRowRender,
      onExpand,
    }));
    wrapper.find('ExpandIcon').first().simulate('click');
    expect(onExpand).toBeCalledWith(true, data[0]);
    wrapper.find('ExpandIcon').first().simulate('click');
    expect(onExpand).toBeCalledWith(false, data[0]);
  });

  it('fires onExpandedRowsChange event', () => {
    const onExpandedRowsChange = jest.fn();
    const wrapper = mount(createTable({
      expandedRowRender,
      onExpandedRowsChange,
    }));
    wrapper.find('ExpandIcon').first().simulate('click');
    expect(onExpandedRowsChange).toBeCalledWith([0]);
  });

  it('fires onExpandedRowsChange event when row is removed', () => {
    const onExpandedRowsChange = jest.fn();
    const wrapper = mount(createTable({
      defaultExpandAllRows: true,
      expandedRowRender,
      onExpandedRowsChange,
    }));
    wrapper.setProps({ data: [
      { key: 1, name: 'Jack', age: 28 },
    ] });
    expect(onExpandedRowsChange).toBeCalledWith([1]);
  });

  it('expand row by click', () => {
    const wrapper = mount(createTable({ expandedRowRender }));
    wrapper.find('ExpandIcon').first().simulate('click');
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
