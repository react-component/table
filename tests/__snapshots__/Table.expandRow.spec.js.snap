// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.expand controlled by expandedRowKeys 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left: 0px;"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-expanded"
              />
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-expanded-row rc-table-expanded-row-level-1"
            data-row-key="0-extra-row"
          >
            <td
              class=""
              colspan="2"
            >
              <p>
                extra data
              </p>
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left: 0px;"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand controlled by expandedRowKeys 2`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left: 0px;"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-expanded-row rc-table-expanded-row-level-1"
            data-row-key="0-extra-row"
            style="display: none;"
          >
            <td
              class=""
              colspan="2"
            >
              <p>
                extra data
              </p>
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left: 0px;"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-expanded"
              />
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
          <tr
            class="rc-table-expanded-row rc-table-expanded-row-level-1"
            data-row-key="1-extra-row"
          >
            <td
              class=""
              colspan="2"
            >
              <p>
                extra data
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand expand all rows by default 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-expanded"
              />
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-expanded-row rc-table-expanded-row-level-1"
            data-row-key="0-extra-row"
          >
            <td
              class=""
              colspan="2"
            >
              <p>
                extra data
              </p>
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-expanded"
              />
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
          <tr
            class="rc-table-expanded-row rc-table-expanded-row-level-1"
            data-row-key="1-extra-row"
          >
            <td
              class=""
              colspan="2"
            >
              <p>
                extra data
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand expand row by click 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left: 0px;"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-expanded"
              />
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-expanded-row rc-table-expanded-row-level-1"
            data-row-key="0-extra-row"
          >
            <td
              class=""
              colspan="2"
            >
              <p>
                extra data
              </p>
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left: 0px;"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand expand rows by defaultExpandedRowKeys 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-expanded"
              />
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
          <tr
            class="rc-table-expanded-row rc-table-expanded-row-level-1"
            data-row-key="1-extra-row"
          >
            <td
              class=""
              colspan="2"
            >
              <p>
                extra data
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand renders a custom icon 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <a
                class="expand-row-icon"
              >
                <i
                  class="some-class"
                />
              </a>
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <a
                class="expand-row-icon"
              >
                <i
                  class="some-class"
                />
              </a>
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand renders expand icon as cell 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col
            class="rc-table-expand-icon-col"
          />
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class="rc-table-expand-icon-th"
              rowspan="1"
              title=""
            />
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class="rc-table-row-expand-icon-cell"
            >
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
            </td>
            <td
              class=""
            >
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class="rc-table-row-expand-icon-cell"
            >
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
            </td>
            <td
              class=""
            >
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand renders expand icon to the specify column 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              Lucy
            </td>
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              27
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              Jack
            </td>
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand renders expand row correctly 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand renders expend row class correctly 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-expanded"
              />
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-expanded-row expand-row-test-class-name rc-table-expanded-row-level-1"
            data-row-key="0-extra-row"
          >
            <td
              class=""
              colspan="2"
            >
              <p>
                extra data
              </p>
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand renders fixed column correctly 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-scroll"
    >
      <div
        class="rc-table-body"
      >
        <table
          class=""
        >
          <colgroup>
            <col />
            <col />
            <col />
          </colgroup>
          <thead
            class="rc-table-thead"
          >
            <tr>
              <th
                class="rc-table-fixed-columns-in-body"
              >
                Name
              </th>
              <th
                class=""
              >
                Age
              </th>
              <th
                class="rc-table-fixed-columns-in-body"
              >
                Gender
              </th>
            </tr>
          </thead>
          <tbody
            class="rc-table-tbody"
          >
            <tr
              class="rc-table-row rc-table-row-level-0"
              data-row-key="0"
            >
              <td
                class="rc-table-fixed-columns-in-body"
              >
                <span
                  class="rc-table-row-indent indent-level-0"
                  style="padding-left:0px"
                />
                <span
                  class="rc-table-row-expand-icon rc-table-row-collapsed"
                />
                Lucy
              </td>
              <td
                class=""
              >
                27
              </td>
              <td
                class="rc-table-fixed-columns-in-body"
              >
                F
              </td>
            </tr>
            <tr
              class="rc-table-row rc-table-row-level-0"
              data-row-key="1"
            >
              <td
                class="rc-table-fixed-columns-in-body"
              >
                <span
                  class="rc-table-row-indent indent-level-0"
                  style="padding-left:0px"
                />
                <span
                  class="rc-table-row-expand-icon rc-table-row-collapsed"
                />
                Jack
              </td>
              <td
                class=""
              >
                28
              </td>
              <td
                class="rc-table-fixed-columns-in-body"
              >
                M
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="rc-table-fixed-left"
    >
      <div
        class="rc-table-body-outer"
        style="-webkit-transform:translate3d (0, 0, 0)"
      >
        <div
          class="rc-table-body-inner"
        >
          <table
            class="rc-table-fixed"
          >
            <colgroup>
              <col />
            </colgroup>
            <thead
              class="rc-table-thead"
            >
              <tr>
                <th
                  class=""
                >
                  Name
                </th>
              </tr>
            </thead>
            <tbody
              class="rc-table-tbody"
            >
              <tr
                class="rc-table-row rc-table-row-level-0"
                data-row-key="0"
              >
                <td
                  class=""
                >
                  <span
                    class="rc-table-row-indent indent-level-0"
                    style="padding-left:0px"
                  />
                  <span
                    class="rc-table-row-expand-icon rc-table-row-collapsed"
                  />
                  Lucy
                </td>
              </tr>
              <tr
                class="rc-table-row rc-table-row-level-0"
                data-row-key="1"
              >
                <td
                  class=""
                >
                  <span
                    class="rc-table-row-indent indent-level-0"
                    style="padding-left:0px"
                  />
                  <span
                    class="rc-table-row-expand-icon rc-table-row-collapsed"
                  />
                  Jack
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      class="rc-table-fixed-right"
    >
      <div
        class="rc-table-body-outer"
        style="-webkit-transform:translate3d (0, 0, 0)"
      >
        <div
          class="rc-table-body-inner"
        >
          <table
            class="rc-table-fixed"
          >
            <colgroup>
              <col />
            </colgroup>
            <thead
              class="rc-table-thead"
            >
              <tr>
                <th
                  class=""
                >
                  Gender
                </th>
              </tr>
            </thead>
            <tbody
              class="rc-table-tbody"
            >
              <tr
                class="rc-table-row rc-table-row-level-0"
                data-row-key="0"
              >
                <td
                  class=""
                >
                  F
                </td>
              </tr>
              <tr
                class="rc-table-row rc-table-row-level-0"
                data-row-key="1"
              >
                <td
                  class=""
                >
                  M
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table.expand renders nested data correctly 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-spaced"
              />
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table.expand renders tree row correctly 1`] = `
<div
  class="rc-table rc-table-scroll-position-left"
>
  <div
    class="rc-table-content"
  >
    <div
      class="rc-table-body"
    >
      <table
        class=""
      >
        <colgroup>
          <col />
          <col />
        </colgroup>
        <thead
          class="rc-table-thead"
        >
          <tr>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
          </tr>
        </thead>
        <tbody
          class="rc-table-tbody"
        >
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="0"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-collapsed"
              />
              Lucy
            </td>
            <td
              class=""
            >
              27
            </td>
          </tr>
          <tr
            class="rc-table-row rc-table-row-level-0"
            data-row-key="1"
          >
            <td
              class=""
            >
              <span
                class="rc-table-row-indent indent-level-0"
                style="padding-left:0px"
              />
              <span
                class="rc-table-row-expand-icon rc-table-row-spaced"
              />
              Jack
            </td>
            <td
              class=""
            >
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
